In [1]:
#Load Packages 

import numpy as np
import pandas as pd

In [2]:
#Read Data

data = pd.read_excel('gender_classification.xlsx')

In [3]:
data.head()

long_hair  forehead_width_cm  forehead_height_cm  nose_wide  nose_long  \
0          1               11.8                 6.1          1          0   
1          0               14.0                 5.4          0          0   
2          0               11.8                 6.3          1          1   
3          0               14.4                 6.1          0          1   
4          1               13.5                 5.9          0          0   

   lips_thin  distance_nose_to_lip_long  gender  
0          1                          1    Male  
1          1                          0  Female  
2          1                          1    Male  
3          1                          1    Male  
4          0                          0  Female

In [4]:
data.shape

(5001, 8)

In [5]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5001 entries, 0 to 5000
Data columns (total 8 columns):
 #   Column                     Non-Null Count  Dtype  
---  ------                     --------------  -----  
 0   long_hair                  5001 non-null   int64  
 1   forehead_width_cm          5001 non-null   float64
 2   forehead_height_cm         5001 non-null   float64
 3   nose_wide                  5001 non-null   int64  
 4   nose_long                  5001 non-null   int64  
 5   lips_thin                  5001 non-null   int64  
 6   distance_nose_to_lip_long  5001 non-null   int64  
 7   gender                     5001 non-null   object 
dtypes: float64(2), int64(5), object(1)
memory usage: 312.7+ KB


In [6]:
data.describe()

long_hair  forehead_width_cm  forehead_height_cm    nose_wide  \
count  5001.000000        5001.000000         5001.000000  5001.000000   
mean      0.869626          13.181484            5.946311     0.493901   
std       0.336748           1.107128            0.541268     0.500013   
min       0.000000          11.400000            5.100000     0.000000   
25%       1.000000          12.200000            5.500000     0.000000   
50%       1.000000          13.100000            5.900000     0.000000   
75%       1.000000          14.000000            6.400000     1.000000   
max       1.000000          15.500000            7.100000     1.000000   

         nose_long    lips_thin  distance_nose_to_lip_long  
count  5001.000000  5001.000000                5001.000000  
mean      0.507898     0.493101                   0.498900  
std       0.499988     0.500002                   0.500049  
min       0.000000     0.000000                   0.000000  
25%       0.000000     0.000000                   0.000000  
50%       1.000000     0.000000                   0.000000  
75%       1.000000     1.000000                   1.000000  
max       1.000000     1.000000                   1.000000

long_hair: yes(1), no(0)
forehead_width_cm: range 11.4 to 15.5
forehead_height_cm: range 5.1 to 7.1
nose_wide: yes(1), no(0)
nose_long: yes(1), no(0)
lips_thin: yes(1), no(0)
distance_nose_to_lip_long: yes(1), no(0)
gender: Male, Female

In [7]:
data['gender'].unique()

array(['Male', 'Female'], dtype=object)

In [8]:
y = data[['gender']].values
y

array([['Male'],
       ['Female'],
       ['Male'],
       ...,
       ['Female'],
       ['Female'],
       ['Male']], dtype=object)

In [9]:
x = data[['long_hair', 'forehead_width_cm', 'forehead_height_cm', 'nose_wide', 'nose_long', 'lips_thin', 'distance_nose_to_lip_long']].values
x

array([[ 1. , 11.8,  6.1, ...,  0. ,  1. ,  1. ],
       [ 0. , 14. ,  5.4, ...,  0. ,  1. ,  0. ],
       [ 0. , 11.8,  6.3, ...,  1. ,  1. ,  1. ],
       ...,
       [ 1. , 12.9,  5.7, ...,  0. ,  0. ,  0. ],
       [ 1. , 13.2,  6.2, ...,  0. ,  0. ,  0. ],
       [ 1. , 15.4,  5.4, ...,  1. ,  1. ,  1. ]])

In [10]:
from sklearn.preprocessing import LabelEncoder

In [11]:
#Change the gender column to a numerical column by using Label Encoder

label_encoder_y = LabelEncoder()
y[:,0] = label_encoder_y.fit_transform(y[:,0])
y

array([[1],
       [0],
       [1],
       ...,
       [0],
       [0],
       [1]], dtype=object)

In [12]:
from sklearn.model_selection import train_test_split

In [13]:
#Split the data into a training and testing set

x_train, x_test, y_train, y_test = train_test_split(x, y, test_size = 0.3, random_state = 0)

Since there are features with different units we will scale them by putting them all on a consistent scale. 

In [14]:
from sklearn.preprocessing import StandardScaler

In [15]:
sc_x = StandardScaler()

In [16]:
x_train = sc_x.fit_transform(x_train)

In [17]:
x_test = sc_x.transform(x_test)

In [18]:
x_train

array([[ 0.39771019, -0.78834195, -0.26362776, ...,  0.97963588,
         1.01033916, -1.01091673],
       [ 0.39771019,  1.36995033,  0.64939139, ...,  0.97963588,
         1.01033916, -1.01091673],
       [ 0.39771019, -0.06891119,  0.28418373, ..., -1.02078744,
         1.01033916,  0.98920116],
       ...,
       [ 0.39771019,  1.01023495,  0.46678756, ...,  0.97963588,
        -0.98976664, -1.01091673],
       [-2.51439371, -1.05812848,  1.56241054, ...,  0.97963588,
         1.01033916,  0.98920116],
       [ 0.39771019,  1.01023495, -0.44623159, ..., -1.02078744,
        -0.98976664, -1.01091673]])

In [19]:
x_test

array([[ 0.39771019, -0.6984131 , -1.54185456, ..., -1.02078744,
        -0.98976664, -1.01091673],
       [ 0.39771019, -0.6984131 , -0.44623159, ...,  0.97963588,
         1.01033916,  0.98920116],
       [ 0.39771019,  0.20087535, -0.81143925, ...,  0.97963588,
         1.01033916,  0.98920116],
       ...,
       [ 0.39771019,  1.01023495,  0.46678756, ..., -1.02078744,
        -0.98976664,  0.98920116],
       [ 0.39771019,  0.56059073, -0.81143925, ..., -1.02078744,
        -0.98976664, -1.01091673],
       [-2.51439371, -0.51855541, -0.99404307, ..., -1.02078744,
        -0.98976664,  0.98920116]])